p4-16/psa/examples/psa-example-clone-to-port.p4

```diff
@@ -0,0 +1,160 @@
+    EthernetAddress srcAddr;
+    bit<16>         etherType;
+}
+
+header ipv4_t {
+    bit<4>  version;
+    bit<4>  ihl;
+    bit<8>  diffserv;
+    bit<16> totalLen;
+    bit<16> identification;
+    bit<3>  flags;
+    bit<13> fragOffset;
+    bit<8>  ttl;
+    bit<8>  protocol;
+    bit<16> hdrChecksum;
+    bit<32> srcAddr;
+    bit<32> dstAddr;
+}
+
+struct fwd_metadata_t {
+
+}
+
+struct metadata {
+    fwd_metadata_t fwd_metadata;
+}
+
+struct headers {
+    ethernet_t       ethernet;
+    ipv4_t           ipv4;
+}
+
+parser CommonParser(packet_in buffer,
+                    out headers parsed_hdr,
+                    inout metadata user_meta)
+{
+    state start {
+        transition parse_ethernet;
+    }
+    state parse_ethernet {
+        buffer.extract(parsed_hdr.ethernet);
+        transition select(parsed_hdr.ethernet.etherType) {
+            0x0800: parse_ipv4;
+            default: accept;
+        }
+    }
+    state parse_ipv4 {
+        buffer.extract(parsed_hdr.ipv4);
+        transition accept;
+    }
+}
+
+parser IngressParserImpl(packet_in buffer,
+                         out headers parsed_hdr,
+                         inout metadata user_meta,
+                         in psa_ingress_parser_input_metadata_t istd,
+                         out psa_parser_output_metadata_t ostd)
+{
+    CommonParser() p;
+
+    state start {
+        p.apply(buffer, parsed_hdr, user_meta);
+        transition accept;
+    }
+}
+
+// clone a packet to CPU.
+control ingress(inout headers hdr,
+                inout metadata user_meta,
+                PacketReplicationEngine pre,
+                in  psa_ingress_input_metadata_t  istd,
+                out psa_ingress_output_metadata_t ostd)
+{
+    action do_clone (PortId_t port) {
+        ostd.clone = true;
+        ostd.clone_port = port;
+    }
+    table t() {
+        key = {
+            user_meta : exact;
+        }
+        actions = { do_clone; }
+    }
+
+    apply {
+        t.apply();
+    }
+}
+
+parser EgressParserImpl(packet_in buffer,
+                        out headers parsed_hdr,
+                        inout metadata user_meta,
+                        in psa_egress_parser_input_metadata_t istd,
+                        out psa_parser_output_metadata_t ostd)
+{
+    CommonParser() p;
+
+    state start {
+        p.apply(buffer, parsed_hdr, user_meta);
+        transition accept;
+    }
+}
+
+control egress(inout headers hdr,
+               inout metadata user_meta,
+               BufferingQueueingEngine bqe,
+               in  psa_egress_input_metadata_t  istd,
+               out psa_egress_output_metadata_t ostd)
+{
+    apply { }
+}
+
+control computeChecksum(inout headers hdr, inout metadata meta) {}
+
+control DeparserImpl(packet_out packet, clone cl, in headers hdr) {
+    // Alternatively, we can provide a constructor to Clone().
+    // Clone() clone;
+    apply {
+        cl.emit({hdr.ethernet.srcAddr, hdr.ipv4.srcAddr});
```

Comment:
So, where are these used? I do not see anything special in the egress parser to process these...